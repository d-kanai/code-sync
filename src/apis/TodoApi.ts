export async function getTodoList() {
  // await api call
  console.log('api call: getTodoList')
  return [
    { id: 1, name: "todo item" },
    { id: 2, name: "todo item2" },
    { id: 3, name: "clean room" }
  ];
}

export async function updateTodoStatus(id:number, status:boolean) {
  console.log('api call: updateTodoStatus', id, status)
}
