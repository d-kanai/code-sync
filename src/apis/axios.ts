import axios from "axios";

//@IMPROVE: use env var
export const http = axios.create({
  //baseURL: "http://54.199.13.127:9000",
  baseURL: "http://localhost:9000",
  headers: {
    "Content-Type": "application/json;charset=utf-8",
    "Access-Control-Allow-Origin": "*",
  },
  timeout: 10000,
});
