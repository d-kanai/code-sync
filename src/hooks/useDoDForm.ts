import * as yup from "yup";
import { useField, useForm } from "vee-validate";

export const useDoDForm = () => {
  const { handleSubmit, errors } = useForm({
    validationSchema: yup.object({
      name: yup.string().required("name is required")
    }),
  });
  const name = useField("name").value
  return { handleSubmit, errors, name }
}