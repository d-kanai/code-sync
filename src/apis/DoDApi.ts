import axios from "axios"

export async function createDoD(params: Object): Promise<String> {
  await axios.post("/dods", params)
  console.log('api call: createDoD', params)
  return "success"
}

export async function findDoDList(): Promise<DoDList> {
  console.log('api call: findDoDList')
  axios.defaults.baseURL = 'http://localhost:9000';
  axios.defaults.headers.post['Content-Type'] = 'application/json;charset=utf-8';
  axios.defaults.headers.post['Access-Control-Allow-Origin'] = '*';
  const dodList = await axios.get("/dods")
  console.log(dodList.data);
  return {
    items: 
      dodList.data
  }
  // return {
  //   items: [
  //     {
  //       name: "Long Method",
  //       data: [
  //         { id: 1, date: "2020-01-01", value: 10, comment: 'something'},
  //         { id: 2, date: "2020-01-02", value: 20, comment: 'something'},
  //         { id: 3, date: "2020-01-03", value: 30, comment: 'something'},
  //       ]
  //     },
  //     {
  //       name: "Coverage",
  //       data: [
  //         { id: 1, date: "2020-01-01", value: 81, comment: 'something'},
  //         { id: 2, date: "2020-01-02", value: 80, comment: 'something'},
  //         { id: 3, date: "2020-01-03", value: 90, comment: 'something'},
  //       ]
  //     }
  //   ]
  // } as DoDList
}


export interface DoDDataRecord {
  id: number
  date: string
  value: number
  comment: string
}

export interface DoD {
  name: string
  data: DoDDataRecord[]
}

export interface DoDList {
  items: DoD[]
}