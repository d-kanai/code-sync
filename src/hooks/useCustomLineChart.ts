import { computed, ref } from "vue";
import { useLineChart } from "vue-chart-3";
import { ChartData } from "chart.js";
import { findDoDItems, DoDItems } from "@/apis/DoDApi"

export const useCustomLineChart = () => {
  const rawData = ref({items: []} as DoDItems);

  const chartData = computed(() => {
    const a = {
      labels: rawData.value.items[0]?.data.map(row=>row.date),
      datasets: [ {
         label: rawData.value.items[0]?.name,
         data: rawData.value.items[0]?.data.map(row=>row.value)
      }],
    }
    const { lineChartProps } = useLineChart({ chartData: a});
    console.log(lineChartProps.value)
    return lineChartProps.value
  });

  const load = async () => {
    rawData.value = await findDoDItems()
    return
  }
  load()

  return {chartData, rawData}
}