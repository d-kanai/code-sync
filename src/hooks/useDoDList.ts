import { findDoDList, DoDList } from "@/apis/DoDApi";
import { ref } from "vue";

export const useDoDList = () => {
  const dodList = ref({ items: [] } as DoDList);
  const load = async () => {
    dodList.value = await findDoDList();
  };
  load();
  return dodList;
};
