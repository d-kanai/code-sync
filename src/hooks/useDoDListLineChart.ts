import { computed, ref } from "vue";
import { useLineChart } from "vue-chart-3";
import { findDoDList, DoDList } from "@/apis/DoDApi";

export const useDoDListLineChart = () => {
  const rawData = ref({ items: [] } as DoDList);

  const chartDataList = computed(() => {
    return rawData.value.items.map((item) => {
      let chartData:any
      if(!item.data) {
        chartData = {}
      } else {
        chartData = {
          labels: item.data.map((row) => row.date),
          datasets: [
            {
              label: item.name,
              data: item.data.map((row) => row.value),
            },
          ],
        };
      }
      const { lineChartProps } = useLineChart({ chartData: chartData });
      return lineChartProps.value;
    });
  });

  const load = async () => {
    rawData.value = await findDoDList();
  };
  load();

  return { chartDataList };
};
