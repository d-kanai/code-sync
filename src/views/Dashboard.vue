<template>
  <div>
    <LineChart v-bind="lineChartProps" />
  </div>
</template>

<script lang='ts'>
import { LineChart } from "vue-chart-3"
import { Chart, registerables } from "chart.js"
import { useCustomLineChart } from "@/hooks/useCustomLineChart"

Chart.register(...registerables);

export default {
  name: "App",
  components: { LineChart },
  setup() {
    const lineChartProps = useCustomLineChart()
    const lineChartProps2 = useCustomLineChart()
    return { lineChartProps, lineChartProps2 }
  },
};
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
