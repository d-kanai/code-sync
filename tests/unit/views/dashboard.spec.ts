import { useDoDItemsLineChart } from "@/hooks/useDoDItemsLineChart";
import 'jest-canvas-mock';
import * as api from "@/apis/DoDApi";
import { DoDItems } from "@/apis/DoDApi";
import { flushPromises } from "@vue/test-utils";
import { mountWithFlushPromise } from "@/../tests/unit/helper"

import Dashboard from "@/views/Dashboard.vue"

function mockDoDItemsApi() {
  const response = {
    items: [
      {
        name: "Long Method Mock",
        data: [
          { id: 1, date: "2020-01-01", value: 10, comment: 'something'},
          { id: 2, date: "2020-01-02", value: 20, comment: 'something'},
          { id: 3, date: "2020-01-03", value: 30, comment: 'something'},
        ]
      },
      {
        name: "Coverage Mock",
        data: [
          { id: 1, date: "2020-01-01", value: 81, comment: 'something'},
          { id: 2, date: "2020-01-02", value: 80, comment: 'something'},
          { id: 3, date: "2020-01-03", value: 90, comment: 'something'},
        ]
      }
    ]
  } as DoDItems
  jest.spyOn(api, "findDoDItems").mockResolvedValueOnce(response);
  return response
}

describe("useDoDItemsLineChart", () => {
  it("snapshot", async () => {
    //given
    mockDoDItemsApi()
    //when
    const wrapper = await mountWithFlushPromise(Dashboard);
    //then
    expect(wrapper.element).toMatchSnapshot();
  })
  it("should find computed ChartData From DoDItemsAPI", async () => {
    //given
    mockDoDItemsApi()
    //when
    const actual = useDoDItemsLineChart()
    await flushPromises();
    //then
    expect(actual.chartDataList.value.length).toEqual(2)
    expect(actual.chartDataList.value[0].chartData.datasets[0].label).toEqual('Long Method Mock')
    expect(actual.chartDataList.value[1].chartData.datasets[0].label).toEqual('Coverage Mock')
  });
});
