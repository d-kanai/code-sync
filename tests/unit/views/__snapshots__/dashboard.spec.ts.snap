// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDoDItemsLineChart snapshot 1`] = `
<div>
  
  <div>
    <div
      class=""
      style="max-width: 100%; position: relative;"
    >
      <canvas
        height="0"
        id="line-chart"
        style="max-width: 100%; max-height: 100%; display: block; box-sizing: border-box; height: 0px; width: 0px;"
        width="0"
      />
    </div>
  </div>
  <div>
    <div
      class=""
      style="max-width: 100%; position: relative;"
    >
      <canvas
        height="0"
        id="line-chart"
        style="max-width: 100%; max-height: 100%; display: block; box-sizing: border-box; height: 0px; width: 0px;"
        width="0"
      />
    </div>
  </div>
  
</div>
`;
